R/if_switch_linter.R

```diff
@@ -0,0 +1,83 @@
+#' Require usage of switch() over repeated if/else blocks
+#'
+#' [switch()] statements in R are used to delegate behavior based
+#'   on the value of some input scalar string, e.g.
+#'   `switch(x, a = 1, b = 3, c = 7, d = 8)` will be one of
+#'   `1`, `3`, `7`, or `8`, depending on the value of `x`.
+#'
+#' This can also be accomplished by repeated `if`/`else` statements like
+#'   so: `if (x == "a") 1 else if (x == "b") 2 else if (x == "c") 7 else 8`
+#'   (implicitly, the last `else` assumes x only takes 4 possible values),
+#'   but this is more cluttered and slower (note that `switch()` takes the same
+#'   time to evaluate regardless of the value of `x`, and is faster even
+#'   when `x` takes the first value (here `a`), and that the `if`/`else`
+#'   approach is roughly linear in the number of conditions that need to
+#'   be evaluated, here up to 3 times).
+#'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "if (x == 'a') 1 else if (x == 'b') 2 else 3",
+#'   linters = if_switch_linter()
+#' )
+#'
+#' # okay
+#' lint(
+#'   text = "if (x == 'a') 1 else if (x == 'b' & y == 2) 2 else 3",
```

Comment:
Shouldn't this example use `switch()`?

Reply:
note the condition on y

Reply:
No, I saw that, but it seems odd to provide "okay" example that doesn't use `switch()` in examples for a linter that vouches for using `switch()`. 

Maybe we can include both variants of these okay examples?